url30d916877f_shylux_shyview_tgz-pJ8-org_json_TestJ8/src/shyview/Picturehandler.java

```diff
@@ -0,0 +1,926 @@
+		repaint();
+	}
+	
+	public synchronized void addLists(List<IPicList> lists) {
+		picListList.addAll(lists);
+		redrawlists();
+	}
+	
+	public boolean imageUpdate( ImageObserver imageObserver ) {
+		repaint();
+		return true;
+	}
+
+	public int count() {
+		int value = 0;
+		for (IPicList list: picListList) {
+			value += list.size();
+		}
+		return value;
+	}
+	
+	public void setTimerdelay(int delay) {
+		pref.putInt("timer_delay", delay);
+		this.picChangeTimer.setInitialDelay(delay);
+	}
+	public int getTimerdelay() {
+		return this.picChangeTimer.getDelay();
+	}
+	
+	public void startTimer() {
+		this.picChangeTimer.start();
+	}
+	public void toggleTimer() {
+		if (this.picChangeTimer.isRunning()) this.picChangeTimer.stop();
+		else this.picChangeTimer.start();
+	}
+	public boolean isTimerRunning() {
+		return this.picChangeTimer.isRunning();
+	}
+	
+	public IPicList getList(String listname) {
+		for (IPicList list: picListList) {
+			String listename = list.getName();
+			if (listename.equals(listname)) return list;
+		}
+		IPicList newlist = new PictureList(listname);		
+		picListList.add(newlist);	
+		return newlist;
+	}
+	
+	public void paint(Graphics g) {
+		super.paint(g);
+		
+		if (picListIt.hasCurrent())
+			TitleInformer.getInstance().update(picListIt.current());
+	
+		int position[] = new int[2];
+		position[0] = 1;
+		position[1] = 1;
+		Image image = this.defaultimage;
+
+		if (count() != 0) {
+			//Load image
+			TitleInformer.getInstance().pushProcess("Loading picture");
+			try {
+				
+				image = acpic().getPicture();
+			} catch (StillLoadingException e) {
+				System.out.println(System.currentTimeMillis()+" "+ acpic().getName() +" still loading..");
+				repaint(); // recursive. stack overflow possible.
+			} catch (FileNotFoundException e1) {
+				System.out.println("file not found");
+				e1.printStackTrace();
+				picListIt.current().remove(acpic());
+				TitleInformer.getInstance().clear();
+				if (this.errorimage == null) return;
+				image = this.errorimage;
+			}
+		}
+		
+		TitleInformer.getInstance().pushProcess("Processing");
+		
+		Graphics2D g2 = (Graphics2D)g;
+		BufferedImage output = Picturehandler.toBufferedImage(image);
+		
+		
+		//normal oder gedreht?
+		if (this.rotation == 0 || this.rotation == 180) {
+			
+			double picwidth = image.getWidth(this);
+			double picheight = image.getHeight(this);
+			double boxwidth = this.getWidth();
+			double boxheight = this.getHeight();
+			double pic = (double)picwidth / (double)picheight;
+			double box = (double)boxwidth / (double)boxheight;
+			int width = 0;
+			int height = 0;
+			int top = 0;
+			int left = 0;
+		
+			if (box>pic) {
+				height = (int) boxheight;
+				width = (int) (boxheight / picheight * picwidth);
+				left = (int) ((boxwidth - width) / 2);
+				switch (position[0]) {
+				case 0:
+					left = 0;
+					break;
+				case 1:
+					left = (int) ((boxwidth - width) / 2);
+					break;
+				case 2:
+					left = (int) boxwidth - width;
+				}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 319:328, 332:341, 375:384, 388:397.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.